pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -394,18 +403,34 @@ func (r *Reconciler) patchStatus(ctx context.Context, app *v1beta1.Application,
 func (r *Reconciler) patchStatusWithRetryOnConflict(ctx context.Context, app *v1beta1.Application, phase common.ApplicationPhase) error {
 	app.Status.Phase = phase
 	updateObservedGeneration(app)
-	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		status := app.Status.DeepCopy()
-		if err := r.Client.Get(ctx, client.ObjectKeyFromObject(app), app); err != nil {
-			return errors.WithMessage(err, "failed to get application while patching status")
-		}
-		app.Status = *status
-		err := r.Status().Patch(ctx, app, client.Merge)
-		if err != nil {
-			return errors.WithMessage(err, "failed to re-patch status")
-		}
-		return nil
-	})
+
+	status := app.Status.DeepCopy()
+	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(app), app); err != nil {
+		return errors.WithMessage(err, "failed to get application while patching status")
+	}
+	app.Status = *status
+	err := r.Status().Patch(ctx, app, client.Merge)
+	if err != nil {
+		return errors.WithMessage(err, "failed to re-patch status")
+	}
+	return nil
+}
+
+// Note: Only operations that must override the status should use this function, it should only focus on workflow operations by now.
+func (r *Reconciler) updateStatusWithRetryOnConflict(ctx context.Context, app *v1beta1.Application, phase common.ApplicationPhase) error {
```

Comment:
the function name and the explanation should be changed as we didn't use retryOnConflict